lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -6610,8 +6646,8 @@ void addPassesForSPIRV(ModulePassManager &PassMgr,
   PassMgr.addPass(PreprocessMetadataPass());
   PassMgr.addPass(SPIRVLowerOCLBlocksPass());
   PassMgr.addPass(OCLToSPIRVPass());
-  PassMgr.addPass(SPIRVRegularizeLLVMPass());
   PassMgr.addPass(SPIRVLowerConstExprPass());
+  PassMgr.addPass(SPIRVRegularizeLLVMPass());
```

Comment:
In SPIRVRegularizeLLVMPass, we have some logic where we try to replace a SPIRV Builtin global constant with a variable in local memory. It helps to have the LowerConstExpr pass run before the regularize pass .

Thanks